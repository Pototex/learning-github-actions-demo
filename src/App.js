import React, { useCallback } from 'react';
import './App.css';
import ParagraphView from './components/parapraph/ParagraphView';

const TitleComponent = React.memo(({index, sum, name}) => {
  const [color, setColor] = React.useState('#ffffff')

  const handleColorChange = () => {
    setColor(color === '#ffffff' ? '#f5b5d3' : '#ffffff')
  }

  return (
    <div 
      className="Title-box"
      type="button"
      style={{ backgroundColor: color}}
      onClick={() => handleColorChange()}
    >
      <span>{name} {sum(index)}</span>
    </div>
  )
})



export default function App() {
  const [titles, setTitles] = React.useState([])
  const [order, setOrder] = React.useState(false)
  
  React.useEffect(
    () => {
      let array = new Array(45)
      for(let index = 0; index <= 45; index++) {
        array[index] = index
      }
      setTitles(array)
    },
    []
  )

  const reverse = () => {
    const sorted = titles.sort((a, b) => order ? a - b : b - a)
    setTitles(sorted)
    setOrder(!order)
  } 

  const sum = (index) => {
    return index + 1
  }

  const memoSum = useCallback(sum, [])

  return (
    <div className="App">
      <h1>Es jueves!!!</h1>

      <img src={"very-cold-beer-every.jpg"} alt="Cold beer"/>

      <ParagraphView label={"La mejor música instrumental"}/>

      <ParagraphView label={"La mejor música instrumental"}/>

      <div className="titles">
        {
          titles.map((value) => (
            <TitleComponent
              key={`title-${value}`} 
              index={value}
              sum={memoSum}
              name={`Beer`}>
            </TitleComponent>
          ))
        }
      </div>

      <div>
        <button onClick={() => reverse()}>Change order</button>
      </div>
    </div>
  );
}