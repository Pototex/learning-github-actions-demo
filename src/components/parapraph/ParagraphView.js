import React from 'react';
import '../../App.css';

const ParagraphView = React.memo(({label}) => {
  return (
    <p className="Paragraph-text">
      {label}
    </p>
  )
})

export default ParagraphView