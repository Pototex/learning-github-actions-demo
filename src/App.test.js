import React from 'react'
import App from './App'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import ParagraphView from './components/parapraph/ParagraphView'

const Component = (props) => 
  <ParagraphView/>

describe('Happy App', () => {
  it('today should be friday', () => {
    const wrapper = shallow(<App />)
    const h1 = wrapper.find("h1")
    const result = h1.text()
    expect(result).toBe('Es viernes!!!')
  })

  it('render correctly ParagraphComponent', () => {
    const ParagraphComponent = renderer.create(<Component/>).toJSON()
    expect(ParagraphComponent).toMatchSnapshot()
  })

  it('should render ParagraphComponent', () => {
    const wrapper = shallow(<App />)
    const p = wrapper.find(ParagraphView)
    expect(p.exists()).toBe(true)
  })
})

