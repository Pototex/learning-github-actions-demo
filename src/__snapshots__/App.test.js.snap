// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Happy App render correctly ParagraphComponent 1`] = `
<p
  className="Paragraph-text"
/>
`;
